### PR TITLE
LPS-131777 call getStaticResourceUrl to get timestamp and perform correct redeploys to cdn's

### PR DESCRIPTION
**Pull request by jgarcialr**
Hello Frontend Team,

Description of the issue:

When you use `<@liferay.js file_name>` or `<@liferay.css file_name>` macros, those imports doesn't have a timestamp.

This is causing that when you redeploy the theme, files imported with those macros, doesn't always show latest version and causes problems with some cdn's (pex dxp-cloud cdn)
 
Things I've tried.

Created a new theme.

Inside portal-normal.ftl import two different js files with the macro and with other mechanism provided by liferay.

1. Using the macro:
 `<@liferay.js file_name="${javascript_folder}/unav/unav72.js"/>`
 
2. Using portalUtil.getStaticResourceURL

```
<#if themeDisplay??>
 <#assign
 fundraising_carousel = htmlUtil.escape(portalUtil.getStaticResourceURL(request, "${themeDisplay.getPathThemeJavaScript()}/unav/fundraising_carousel.js"))>
 <script src="${fundraising_carousel}" type="text/javascript"></script>
```

For the first case, a <script tag is created. src attribute, contains the filename, without any parameters.
For the second case, the source of the script tag has a timestamp and browserId parameters.

Test plan

    Deploy [my-liferay-theme](https://issues.liferay.com/secure/attachment/404645/my-liferay-theme.war) inside a portal bundle
    Create a new Site Site1
    Set my-liferay-theme as theme for this site
    Navigate to Site Builder > Pages
    Created a new Content Page
    Publish the page
    Navigate to the page you just published
    In network panel, just second request has parameters (fundraising_carousel.js) request has parameters.

After applying this change both requests have parameters and those parameters are updated with each redeployment of the theme

### CODE REVIEW

File: modules/apps/portal-template/portal-template-freemarker/src/main/resources/FTL_liferay.ftl [hunk 1]
```diff
@@ -66,9 +66,9 @@ LPS-30525.
 	file_name
 >
 	<#if file_name == css_main_file>
-		<link class="lfr-css-file" href="${file_name}" id="mainLiferayThemeCSS" rel="stylesheet" type="text/css" />
+		<link class="lfr-css-file" href="${htmlUtil.escape(portalUtil.getStaticResourceURL(request,file_name))}" id="mainLiferayThemeCSS" rel="stylesheet" type="text/css" />
```
Comment (wincent) [inline modules/apps/portal-template/portal-template-freemarker/src/main/resources/FTL_liferay.ftl:69]: Question about formatting: do we expect a space after the `request,`?

**Review by wincent — COMMENTED** [review]
Freemarker templates and static resource generation aren't my fields of expertise, but this looks fine to me. Let's get a second opinion on this from somebody who actually _knows_ — maybe @izaera or @markocikos.

File: modules/apps/portal-template/portal-template-freemarker/src/main/resources/FTL_liferay.ftl (hunk 1 above)
Comment (nhpatt) [reply modules/apps/portal-template/portal-template-freemarker/src/main/resources/FTL_liferay.ftl:69]: Yes, as you can see in line 23: 

`htmlUtil.escape(portalUtil.getStaticResourceURL(request, "${themeDisplay.getPathThemeCss()}/main.css"))` 

SF in freemarker is almost useless, so Brian usually fixes manually all of these.

Comment (wincent) [reply modules/apps/portal-template/portal-template-freemarker/src/main/resources/FTL_liferay.ftl:69]: So, up to you @jgarcialr: you can either add another commit with the formatting fixes just to be safe, or go for the coin toss that Brian will do a "please resend" or a "see `1234abcd`" 😁

Comment (jgarcialr) [reply modules/apps/portal-template/portal-template-freemarker/src/main/resources/FTL_liferay.ftl]: Thanks for the tips :grin: 
I'll go with the safe bet.

**Comment by izaera** [comment]
Not horrible ™

Changing things so deep in the infra is always risky, but this seems harmless and the fix makes sense to me. We can forward :+1:

**Comment by izaera** [comment]
ci:forward

**Pull request closed without merging**